src/Tools/dotnet-monitor/Egress/S3/S3Storage.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Amazon.Runtime.CredentialManagement;
+using Amazon.Runtime;
+using Amazon.S3;
+using System.Threading;
+using System.Threading.Tasks;
+using System.IO;
+using Amazon.S3.Model;
+using Amazon.S3.Transfer;
+using System.Collections.Generic;
+using System;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.S3
+{
+    internal class S3Storage : IS3Storage
+    {
+        private readonly IAmazonS3 _s3Client;
+        private readonly string _bucketName;
+        private readonly string _objectId;
+        private readonly string _contentType;
+
+        public S3Storage(IAmazonS3 client, string bucketName, string objectId, string contentType)
+        {
+            _s3Client = client;
+            _bucketName = bucketName;
+            _objectId = objectId;
+            _contentType = contentType;
+        }
+
+        public static async Task<IS3Storage> CreateAsync(S3StorageEgressProviderOptions options, EgressArtifactSettings settings, CancellationToken cancellationToken)
+        {
+            AWSCredentials awsCredentials = null;
+            AmazonS3Config configuration = new();
+            // use the specified access key and the secrets taken from configuration or a local file
+            if (!string.IsNullOrEmpty(options.AccessKeyId) && (!string.IsNullOrEmpty(options.SecretAccessKey) || !string.IsNullOrEmpty(options.SecretsAccessKeyFile)))
+            {
+                string secretAccessKeyId;
+                if (!string.IsNullOrEmpty(options.SecretsAccessKeyFile) && File.Exists(options.SecretsAccessKeyFile))
+                    secretAccessKeyId = (await File.ReadAllTextAsync(options.SecretsAccessKeyFile, cancellationToken)).Trim();
```

Comment:
If both `options.SecretAccessKey` and `options.SecretsAccessKeyFile` are provided, it looks like we disregard `options.SecretAccessKey`. Should this be used as a fallback if the contents of `options.SecretsAccessKeyFile` don't succeed as the `secretAccessKeyId`?

Reply:
@kkeirstead generally speaking  SecretsAccessKeyFile case would be less (special cases) compared to SecretAccessKey.  We discussed in the previous version and decided to be either/or, because you don't want to hang around 2 secrets and keep trying which one would work. In-fact after this PR merged I am going to take a stab on the documentation for s3 egress what's the best security polity (which is to use profile or default cred chain)

Reply:
This should be reversed: `SecretAccessKey` should be higher precedence over `SecretsAccessKeyFile` (if this property remains) such that `SecretsAccessKeyFile` is a fallback.